### PR TITLE
SCRUM-1724 create endpoint to generate AGMDiseaseAnnotationDTO

### PR DESCRIPTION
**Pull request by cmpich**

### CODE REVIEW

**Comment by oblodgett** [comment]
Do we not have endpoints for these?

**Comment by cmpich** [comment]
No. we don't. We have an endpoint to load bulk data but I cannot use it as it basically drops and re-loads according to the submitted file.

**Comment by cmpich** [comment]
I need an endpoint to only load one AgmDiseaseAnnotationDTO object.

**Comment by oblodgett** [comment]
And you don't want to use the endpoint for the AgmDiseaseAnnotation object directly?

**Comment by cmpich** [comment]
No, because I don't have all the internal / curie ids.

**Comment by oblodgett** [comment]
I would prefer to not go down the road of using the DTO's. Those are only for ingest data and will go away once a mod moves over. So it would mean that we have 2 create methods for each data type. It would be better to use only one create API call.

File: src/main/java/org/alliancegenome/curation_api/base/services/BaseDTOCrudService.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import org.alliancegenome.curation_api.base.BaseDTO;
+import org.alliancegenome.curation_api.base.dao.BaseEntityDAO;
+import org.alliancegenome.curation_api.base.entity.BaseEntity;
+import org.alliancegenome.curation_api.exceptions.ObjectUpdateException;
+
+
+public abstract class BaseDTOCrudService<E extends BaseEntity, T extends BaseDTO, D extends BaseEntityDAO<E>> extends BaseEntityCrudService<E, D> {
+
+	protected abstract void init();
+	public abstract E upsert(T dto) throws ObjectUpdateException;
```
Comment (oblodgett) [inline src/main/java/org/alliancegenome/curation_api/base/services/BaseDTOCrudService.java:12]: @cmpich if you need to have an ObjectResponse we would need to change this right now they all return just the Entity from the DB.

**Review by oblodgett — APPROVED** [review]

File: src/main/java/org/alliancegenome/curation_api/base/services/BaseDTOCrudService.java (hunk 1 above)
Comment (oblodgett) [reply src/main/java/org/alliancegenome/curation_api/base/services/BaseDTOCrudService.java:12]: `E` -> `ObjectResponse<E>`

**Pull request merged**